webapp/packages/core-localization/src/LocalizationService.test.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { beforeEach, describe, expect, it } from '@jest/globals';
+
+import { DEFAULT_LOCALE } from './DEFAULT_LOCALE.js';
+import { type ILocale } from './ILocale.js';
+import { LocalizationService } from './LocalizationService.js';
+
+describe('LocalizationService', () => {
+  let service: LocalizationService;
+
+  beforeEach(() => {
+    service = new LocalizationService();
+  });
+
+  it('should initialize with default locale', () => {
+    service.register();
+    expect(service.currentLanguage).toBe(DEFAULT_LOCALE.isoCode);
+    expect(service.supportedLanguages.length).toBeGreaterThan(0);
+  });
+
+  it('should set supported languages and default to first supported language', () => {
+    const locales: ILocale[] = [
+      { isoCode: 'de', name: 'German', nativeName: 'Deutsch' },
+      { isoCode: 'es', name: 'Spanish', nativeName: 'Español' },
+    ];
+
+    service.setSupportedLanguages(locales);
+    expect(service.supportedLanguages.length).toBe(2);
+    expect(service.currentLanguage).toBe('de');
+  });
+
+  it('should return default locale if the current language is not supported', () => {
+    service.setLanguage('es');
+    expect(service.currentLanguage).toBe(DEFAULT_LOCALE.isoCode);
+  });
+
+  it('should throw error if there are no supported languages', () => {
+    service.supportedLanguages = [];
+    expect(() => service.currentLanguage).toThrowError('No language is available');
```

Comment:
this seems weird that getter can throw an error. I suggest to add TODO to refactor this logic somewhen in the `LocalizationService.ts`